assets/src/hooks/useNotifications.ts

```diff
@@ -0,0 +1,52 @@
+import { Channel, Socket } from "phoenix"
+import { useContext, useEffect } from "react"
+import { SocketContext } from "../contexts/socketContext"
+import { StateDispatchContext } from "../contexts/stateDispatchContext"
+import { reload } from "../models/browser"
+import { notificationsFromData } from "../models/notificationData"
+import { Notification } from "../realtime.d"
+
+export const useNotifications = (
+  handleNewNotification: (notification: Notification) => void
+): void => {
+  const { socket }: { socket: Socket | undefined } = useContext(SocketContext)
+  const topic: string = "notifications"
+  const event: string = "notifications"
+  const [{ selectedRouteIds }] = useContext(StateDispatchContext)
+
+  useEffect(() => {
+    let channel: Channel | undefined
+
+    if (socket !== undefined) {
+      channel = socket.channel(topic)
+      channel.on(event, ({ data: data }) => {
+        const notifications: Notification[] = notificationsFromData(data)
```

Comment:
@skyqrose in the old implementation it was possible for the data to be `undefined` (i.e. when a client first joined). Is that accounted for here?

Reply:
When the backend notifications channel doesn't send a response to joining the chanel, the default response is `{}`. The old version logged undefined, because it was doing `console.log({}.data)`. `useNotifications()` avoids that by not giving a `channel.receive("ok",)` handler to the join response like `useChannel()` did.